dlib/test/matrix.cpp

```diff
@@ -23,8 +24,27 @@ namespace
     using namespace dlib;
     using namespace std;
 
+    dlib::rand rnd;
+
     logger dlog("test.matrix");
 
+    template <typename type>
+    const matrix<type> rand_sp_banded(long n, long bw)
+    {
+        matrix<type> m = 10 * identity_matrix<type>(n);
```

Comment:
The tests look good.  Make sure to add one for column major layout as well so both code paths are covered though.

Reply:
I have now added a column major layout test.